### PR TITLE
PLANNER-750 Backwards incompatible changes to the API

### PR DESCRIPTION
**Pull request by triceo**

### CODE REVIEW

File: .github/workflows/os_java_ci.yml [hunk 1]
```diff
@@ -5,10 +5,10 @@ name: OptaPlanner CI
 on:
   push:
     branches:
-      - master
+      - future-8.x
   pull_request:
     branches:
-      - master
+      - future-8.x
```
Comment (ge0ffrey) [inline .github/workflows/os_java_ci.yml:11]: These should probably stay master I guess? cc @rsynek
The future-8.x branch doesn't exist in the optaplanner repo, only on your fork.

Comment (rsynek) [reply .github/workflows/os_java_ci.yml:11]: Here, yes. On the fork, it must be the future-8.x branch. I wonder if we just can have both of them (master, future-8.x) in the yaml file until this PR is ready to be merged to master.

Comment (triceo) [reply .github/workflows/os_java_ci.yml:11]: Yes, we can.

Comment (rsynek) [reply .github/workflows/os_java_ci.yml:11]: Actually, we don't necessarily have to: the OS x JDK matrix is available on the small, incremental PRs, while Jenkins PR check and the SonarCloud are available here.

@ge0ffrey I would just leave it as it is and before we merge this PR, we put the master branch back in the yaml file.

Comment (ge0ffrey) [reply .github/workflows/os_java_ci.yml:11]: Sounds good :)

**Pull request merged**